lib/experimental/class-wp-webfonts-utils.php

```diff
@@ -0,0 +1,113 @@
+ *
+ * @package    WordPress
+ * @subpackage WebFonts
+ * @since      X.X.X
+ */
+
+if ( class_exists( 'WP_Webfonts_Utils' ) ) {
+	return;
+}
+
+/**
+ * Utility helpers for the Webfonts API.
+ *
+ * @since X.X.X
+ */
+class WP_Webfonts_Utils {
+
+	/**
+	 * Converts the given font family into a handle.
+	 *
+	 * @since X.X.X
+	 *
+	 * @param string $font_family Font family to convert into a handle.
+	 * @return string|null The font family handle on success. Else, null.
+	 */
+	public static function convert_font_family_into_handle( $font_family ) {
+		if ( ! is_string( $font_family ) || empty( $font_family ) ) {
+			return null;
+		}
+
+		return sanitize_title( $font_family );
+	}
+
+	/**
+	 * Converts the given variation and its font-family into a handle.
+	 *
+	 * @since X.X.X
+	 *
+	 * @param string $font_family The font family's handle for this variation.
+	 * @param array  $variation   An array of variation properties.
+	 * @return string|null The variation handle.
+	 */
+	public static function convert_variation_into_handle( $font_family, array $variation ) {
+		$handle = '';
+		foreach ( array( 'font-weight', 'font-style' ) as $property ) {
+			if ( ! array_key_exists( $property, $variation ) || ! static::is_defined( $variation[ $property ] ) ) {
+				continue;
+			}
+
+			$handle .= ' ' . $variation[ $property ];
+		}
+
+		if ( '' === $handle ) {
+			trigger_error( 'Variant handle could not be determined as font-weight and/or font-style are require' );
+			return null;
+		}
+
+		return sanitize_title( $font_family . $handle );
+	}
+
+	/**
+	 * Gets the font family from the variation.
+	 *
+	 * @since X.X.X
+	 *
+	 * @param array $variation An array of variation properties to search.
+	 * @return string|null The font family if defined. Else, null.
+	 */
+	public static function get_font_family_from_variation( array $variation ) {
+		return static::search_for_font_family( $variation );
+	}
+
+	/**
+	 * Checks if the given input is defined, i.e. meaning is a non-empty string.
+	 *
+	 * @since X.X.X
+	 *
+	 * @param string $input The input to check.
+	 * @return bool True when non-empty string. Else false.
+	 */
+	public static function is_defined( $input ) {
+		return ( is_string( $input ) && ! empty( $input ) );
+	}
+
+	/**
+	 * Searches the variation array to extract the font family.
+	 *
+	 * @since X.X.X
+	 *
+	 * @param array $haystack An array of variation properties to search.
+	 * @return string|null The font family when found. Else, null.
+	 */
+	private static function search_for_font_family( array $haystack ) {
+		if ( array_key_exists( 'fontFamily', $haystack ) ) {
+			$key = 'fontFamily';
+		} elseif ( array_key_exists( 'font-family', $haystack ) ) {
+			$key = 'font-family';
+		} else {
+			trigger_error( 'Font family not found.' );
+			return null;
+		}
+
+		if ( static::is_defined( $haystack[ $key ] ) ) {
+			return $haystack[ $key ];
+		}
+
+		trigger_error( 'Font family not defined in the variation.' );
+		return null;
+	}
```

Comment:
Could we not rewrite this a bit to make it simpler/easier?

```suggestion
	private static function search_for_font_family( array $haystack ) {
		if ( array_key_exists( 'fontFamily', $haystack ) && static::is_defined( $haystack['fontFamily'] ) ) {
			return $haystack['fontFamily'];
		}
		if ( array_key_exists( 'font-family', $haystack ) && static::is_defined( $haystack['font-family'] ) ) {
			return $haystack['font-family'];
		}
		trigger_error( 'Font family not found.' );
		return null;
	}
```

Reply:
Hmm then it repeats the `static::is_defined()`. But I do agree that the PHP notices need work.

- The first one is meant to inform the developer that the property is not defined in the variation. 
- The second one is meant to inform the developer that the font-family's value has to be a non-empty string.

2 different notifications. Both need work 🤣 

Hey @aristath do you think a developer needs this level of detail?

Reply:
I think that 2 notices here may be overkill... 😅 
We can definitely simplify it and use a single notification if undefined or empty.

Reply:
I say leave it as is. I see both notices being useful.

Reply:
How about this? Leave it as is and then get feedback from developers if the extra informational notice is helpful or not. If not, then combine into 1 notice. Thoughts?